### PR TITLE
Add "Set as main scene" on context menu in FileSystem panel

### LINKED ISSUE #189: Add "set as main scene" to the context menu in the Filesystem when RMB clicking on .tscn files
**Describe the project you are working on:**
I'm constantly making small prototype test scenes as tests for my main projects, or if I am unsure how a Godot feature works and I want to test it isolated from anything else.

**Describe the problem or limitation you are having in your project:**
When creating those small test scenes, I usually work on it for a few hours to a few days.
During that time, I have always have to use "play scene" or F6 instead of regular play project.

This fucks with my muscle memory a lot. No matter how hard I try to concentrate I run the project (the main scene) instead of the current scene many times over, closing the wrong running scene, hitting "play scene" again. Often I miss-click a second or sometimes even a third time.
This becomes even more of a problem the more you switch back and forth between your regular projects and test scenes.

When I change back to my regular project definitively, I often have the same problem in reverse.

The solution would seem simple: Just change the main scene in the Project settings, and always hit play!
But usually I try to do these tests as quickly as I can, changing the main scene by clicking through the Project settings a number of times a day to change the main scene back and forth, is something I would really like to avoid. Often I have to come back to these test scenes as I realised I haven't figured out something quite yet as I thought I would.

**Describe how this feature / enhancement will help you overcome this problem or limitation:**
Changing the main scene through the RMB context menu in the filesystem panel would make the process incredibly painless, effortless and quick. No more fucked up muscle memory.

**Show a mock up screenshots/video or a flow diagram explaining how your proposal will work:**
In the Filesystem Panel: RMB click on any .tscn file and leftclick on "Set as main scene"

**Describe implementation detail for your proposal (in code), if possible:**
Impossible.

**If this enhancement will not be used often, can it be worked around with a few lines of script?:**
It would be used often I believe by people who experiment a lot, by people who learn the engine, by people who want to try out different things quickly, by people who have had the same frustration with their muscle memory.

**Is there a reason why this should be core and not an add-on in the asset library?:**
It's too small of a UX improvement to justify a download in the asset library.

### PR DESCRIPTION
**Pull request by volzhs**
close https://github.com/godotengine/godot-proposals/issues/189

main_scene

### CODE REVIEW

File: editor/filesystem_dock.cpp [hunk 1]
```diff
@@ -2138,6 +2151,7 @@ void FileSystemDock::_file_and_folders_fill_popup(PopupMenu *p_popup, Vector<Str
 			if (filenames.size() == 1) {
 				p_popup->add_icon_item(get_icon("Load", "EditorIcons"), TTR("Open Scene"), FILE_OPEN);
 				p_popup->add_icon_item(get_icon("CreateNewSceneFrom", "EditorIcons"), TTR("New Inherited Scene"), FILE_INHERIT);
+				p_popup->add_item(TTR("Set as main scene"), FILE_MAIN_SCENE);
```
Comment (akien-mga) [inline editor/filesystem_dock.cpp]: Should be capitalized as "Set As Main Scene".

Comment (volzhs) [reply editor/filesystem_dock.cpp]: updated

**Comment by Calinou** [comment]
See my comment here: https://github.com/godotengine/godot-proposals/issues/189#issuecomment-547641770

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**